### DISCUSSION TITLE
Documenting your efforts

### DISCUSSION THREAD

**Discussion by Analect** [diegoluisi/argo#208] [Ideas]
@diegoluisi and team,
I've been familiarising myself with your work presented under [DevXP Tech](https://github.com/devxp-tech) and I think it's a worthwhile effort. Based on my own experience in trying to get things working, in the absence of documentation, I think I can add some value by putting in place some preliminary docs. Below is a rough structure of how they might be organised. I can do a mock-up if you like. I like to leverage [Quarto](https://quarto.org/), a powerful markdown-based docs tool. It can work with [github actions](https://github.com/quarto-dev/quarto-actions), so it should fit in with your general approach. I would look to mimic some of the functionality on the main Quarto site (docs below would look a bit like their `Guide`. Their approach to `Help` integrates back into issues/discussion on Github and it would be easy to add `Blog` functionality too. Let me know your thoughts. Colum

```
.
├── Overview/
│   ├── High-level overview of gitops pattern (dev, staging, prod)
│   ├── Out-of-the-box integrations
│   ├── Leveraging Github Action workflows (devxp-tech/.github repo)
│   └── Usage of Backstage scaffolder (devxp-tech/helm-charts repo)
├── Infrastructure Setup/
│   ├── Kubernetes/
│   │   ├── Cluster(s)/load-balancer
│   │   ├── Cert-manager - LetsEncrypt/DNS handling
│   │   ├── Sealed-secrets
│   │   ├── Istio
│   │   └── metrics-server
│   ├── ArgoCD/
│   │   ├── argocd-autopilot (devxp-tech/gitops)
│   │   ├── project structure
│   │   └── clusters
│   ├── Github/
│   │   ├── Organization (preferred)
│   │   ├── SSO Auth for various supporting systems
│   │   ├── Other secrets/tokens (org v repo v environment)
│   │   ├── Github App integration for Backstage
│   │   └── Other workflow integrations - SonarQube, OpenAI, Datree, Trivy
│   └── Core integrations/
│       ├── Argo Rollouts
│       ├── Dex
│       ├── Prometheus
│       ├── Grafana (devxp-tech/grafana-dashboards)
│       ├── Kiali
│       ├── Kyverno
│       ├── SonarQube
│       └── Slack
├── Backstage Setup/
│   ├── Integrations/
│   │   ├── ArgoCD
│   │   ├── Github
│   │   ├── Kubernetes
│   │   ├── Grafana
│   │   └── Techdocs/AWS-S3
│   └── Templates/
│       ├── devxp-app, devxp-infra, tooling (devxp-tech/helm-charts)
│       └── devxp-tech/template-golang
├── System Operation/
│   ├── Lifecycle of an app?
│   ├── Control from Backstage?
│   ├── Managing new short-life clusters / resources
│   └── Managing new workload types	
└── Troubleshooting/
    └── Index back into sections of docs
```

**Comment by diegoluisi** [comment]
HI @Analect 
Yes, sure.. I will create more documentations for GitOps and Backstage.. 
If you want, you can open a PR.. thanks!
